Rendera/include/camera.hpp

```diff
@@ -26,9 +27,14 @@ class Camera{
             focal_length {Focal_length} {}
         
         //returns the ray corresponding to (x,y) on the image
-        ray get_ray(int x, int y){
-            float viewport_Y = (x*1.0/img_width) * port_width - port_width/2;
-            float viewport_Z = port_height/2 - (y*1.0/img_height) * port_height;
+        ray get_ray(int x, int y, std::minstd_rand &random) const{
```

Comment:
Ideally, the `get_ray` function should not need to take a parameter for a random number generator, because that should not be a part of the public API of the class. This randomness business should be handled under the hood, and no one outside the `Camera` class should need to know about it.

One way of achieving this is by having a private static member function which will return a random double between, say, 0 and 1 when called, and then adding this number to the `viewport_X` and `viewport_Y` variables.